### PR TITLE
Fixes #36282: Pre-download packages before stopping services

### PR DESCRIPTION
**Pull request by ehelms**
In upgrade scenarios, update packages using --downloadonly option with dnf to stage the packages without having to stop services. Then stop services and actually perform the update. This will cut down on the downtime during the upgrade process.

### CODE REVIEW

File: definitions/procedures/packages/update.rb [hunk 1]
```diff
@@ -27,7 +27,11 @@ def necessary?
     end
 
     def description
-      "Update package(s) #{@packages.join(', ')}"
+      if @yum_options.include?('--downloadonly')
```
Comment (ekohl) [inline definitions/procedures/packages/update.rb:30]: It surprises me that there's a `yum_options` parameter to the generic `Update` procedure. My immediate question is whether it also applies those on DNF.

File: test/test_helper.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
 Mocha.configure do |c|
   c.strict_keyword_argument_matching = true
 end
-Minitest::Reporters.use!
+Minitest::Reporters.use! [Minitest::Reporters::DefaultReporter.new(:color => true)]
```
Comment (ekohl) [inline test/test_helper.rb:13]: This looks unrelated. Can this be separated, at least in a different commit?

File: definitions/procedures/packages/update.rb (hunk 1 above)
Comment (ehelms) [reply definitions/procedures/packages/update.rb:30]: > It surprises me that there's a `yum_options` parameter to the generic `Update` procedure. My immediate question is whether it also applies those on DNF.

It does, as the DNF package manager inherits the yum one. I tested this on a real system.

File: test/test_helper.rb (hunk 2 above)
Comment (ehelms) [reply test/test_helper.rb:13]: Done.

File: definitions/scenarios/upgrade_to_capsule_6_14.rb [hunk 3]
```diff
@@ -58,7 +57,9 @@ def compose
       add_step(Procedures::Repositories::Setup.new(:version => '6.14'))
       modules_to_enable = ["satellite-capsule:#{el_short_name}"]
       add_step(Procedures::Packages::EnableModules.new(:module_names => modules_to_enable))
-      add_step(Procedures::Packages::UnlockVersions.new)
```
Comment (ekohl) [inline definitions/scenarios/upgrade_to_capsule_6_14.rb:61]: Is unlocking no longer needed?

Comment (ehelms) [reply definitions/scenarios/upgrade_to_capsule_6_14.rb:61]: Unlocking is designed to happen whenever an action requires it, explicitly unlocking here serves no purpose and further wastes time.

**Review by evgeni — APPROVED** [review]
